### PR TITLE
Transmission Add Unmarshalling Validation

### PR DESCRIPTION
**Pull request by nkstanchev**
Addresses #130.

Signed-off-by: Nikolay Stanchev <nstanchev@vmware.com>

### CODE REVIEW

File: models/transmission.go [hunk 1]
```diff
@@ -59,6 +60,74 @@ func (t Transmission) MarshalJSON() ([]byte, error) {
...
+	if a.ID != nil {
+		t.ID = *a.ID
+	}
+	t.Notification = a.Notification
+
+	if a.Receiver != nil {
+		t.Receiver = *a.Receiver
+	}
+	t.Channel = a.Channel
+	t.Status = a.Status
```
Comment (tsconn23) [inline models/transmission.go]: If you're going to go down this validation path, then the value of Status should also be checked to ensure it's in the range of acceptable values. We have an example of that if you look into how OperatingState and AdminState are validated on the Device
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/device.go

**Comment by ernestojeda** [comment]
recheck

File: models/transmission.go (hunk 1 above)
Comment (nkstanchev) [reply models/transmission.go]: I think Notification, Channel and Channel's ChannelType would also need to have a Validate method and unit tests. Didn't fully grasp the volume of the required work but I can't get properly tested support services without this. Should these be in a different PR?

Comment (tsconn23) [reply models/transmission.go]: `Validate the NotificationStatus` as part of this PR and then create an issue for the follow-up validation of the dependent types. Once the issue is created, let me know and I can assign to you

File: models/transmission_status.go [hunk 2]
```diff
@@ -39,13 +40,19 @@ func (as *TransmissionStatus) UnmarshalJSON(data []byte) error {
...
 	return nil
 }
+
+func (as TransmissionStatus) Validate() (bool, error) {
+	_, found := map[string]TransmissionStatus{"FAILED": Failed, "SENT": Sent, "ACKNOWLEDGED": Acknowledged, "TRXESCALATED": Trxescalated}[string(as)]
+	if !found {
+		return false, NewErrContractInvalid(fmt.Sprintf("invalid Transmission Status %q", as))
+	}
+	return true, nil
+}
```
Comment (tsconn23) [inline models/transmission_status.go]: Needs unit test

Comment (tsconn23) [reply models/transmission_status.go]: Also transmission.go::UnmarshalJSON() needs to be covered by unit test. That's where the large drop in coverage reported by codecov.io is coming from.

**Comment by lranjbar** [comment]
recheck

**Comment by tsconn23** [comment]
@nkstanchev Needs go fmt
```
17:06:04  CGO_ENABLED=0 GO111MODULE=on go vet ./...
17:06:26  gofmt -l .
17:06:26  models/transmission_test.go
17:06:26  [ "`gofmt -l .`" = "" ]
17:06:26  make: *** [Makefile:9: test] Error 1
```

**Comment by tsconn23** [comment]
recheck

**Pull request merged**